File: sdkwork-app-sdk-typescript/src/types/document-list-query-request.ts
/** List documents query request */
export interface DocumentListQueryRequest {
  /** Keyword for title/content search */
  keyword?: string;
  /** Folder id */
  folderId?: string;
  /** Disk id */
  diskId?: string;
  /** Whether include archived documents */
  includeArchived?: boolean;
  /** Whether include deleted documents */
  includeDeleted?: boolean;
  /** Whether only return favorited documents */
  favoriteOnly?: boolean;
  /** Sort field, allowed: createdAt, updatedAt, name, size */
  sortField?: string;
  /** Sort order, asc or desc */
  sortOrder?: string;
  /** Page number, starts from 1 */
  pageNum?: number;
  /** Page size */
  pageSize?: number;
}

File: sdkwork-app-sdk-go/types/document_list_query_request.go
package types

// List documents query request
type DocumentListQueryRequest struct {
	Keyword string `json:"keyword"`
	FolderId string `json:"folderId"`
	DiskId string `json:"diskId"`
	IncludeArchived bool `json:"includeArchived"`
	IncludeDeleted bool `json:"includeDeleted"`
	FavoriteOnly bool `json:"favoriteOnly"`
	SortField string `json:"sortField"`
	SortOrder string `json:"sortOrder"`
	PageNum int `json:"pageNum"`
	PageSize int `json:"pageSize"`
}
